### PR TITLE
Redis: respect IDatabase key prefix

### PR DESCRIPTION
**Pull request by skomis-mm**
Fixes [#66](https://github.com/madelson/DistributedLock/issues/66)

### CODE REVIEW

File: DistributedLock.Tests/Infrastructure/TestingSynchronizationStrategy.cs [hunk 1]
```diff
@@ -21,5 +21,6 @@ public abstract class TestingSynchronizationStrategy : IDisposable
         public virtual IDisposable? PrepareForHandleLost() => null;
         public virtual void PrepareForHighContention() { }
         public virtual void Dispose() { }
+        public virtual string? GetLockPrefix() => null;
```
Comment (madelson) [inline DistributedLock.Tests/Infrastructure/TestingSynchronizationStrategy.cs:24]: Why is this needed?

File: DistributedLock.Tests/Infrastructure/TestingLockProvider.cs [hunk 2]
```diff
@@ -19,6 +19,8 @@ public abstract class TestingLockProvider<TStrategy> : ITestingNameProvider, IDi
         public virtual string GetCrossProcessLockType() => this.CreateLock(string.Empty).GetType().Name;
         public virtual void Dispose() => this.Strategy.Dispose();
 
+        public string? GetLockPrefix() => Strategy.GetLockPrefix();
```
Comment (madelson) [inline DistributedLock.Tests/Infrastructure/TestingLockProvider.cs:22]: Why is this needed?

File: DistributedLock.Tests/Infrastructure/Redis/TestingRedisSynchronizationStrategy.cs [hunk 3]
```diff
@@ -22,6 +22,15 @@ public void SetOptions(Action<RedisDistributedSynchronizationOptionsBuilder>? op
             this._options = options;
         }
 
+        public override string? GetLockPrefix()
```
Comment (madelson) [inline DistributedLock.Tests/Infrastructure/Redis/TestingRedisSynchronizationStrategy.cs:25]: Why is this needed?

File: DistributedLock.Tests/Infrastructure/Redis/TestingRedisDatabaseProvider.cs [hunk 4]
```diff
@@ -22,17 +24,35 @@ protected TestingRedisDatabaseProvider(int count)
...
+        {
+        }
+
         // publicly settable so that callers can alter the dbs in use
         public IReadOnlyList<IDatabase> Databases { get; set; }
 
         public virtual string CrossProcessLockTypeSuffix => this.Databases.Count.ToString();
     }
 
+    public interface ITestingRedisWithKeyPrefixDatabaseProvider
```
Comment (madelson) [inline DistributedLock.Tests/Infrastructure/Redis/TestingRedisDatabaseProvider.cs:38]: Why is this needed?

File: DistributedLock.Tests/AbstractTestCases/DistributedLockCoreTestCases.cs [hunk 5]
```diff
@@ -443,7 +443,7 @@ private void CrossProcessAbandonmentHelper(bool asyncWait, bool kill)
             }
         }
 
-        private Command RunLockTaker(TLockProvider engine, params string[] args)
+        private Command RunLockTaker(TLockProvider engine, string lockType, string lockName)
```
Comment (madelson) [inline DistributedLock.Tests/AbstractTestCases/DistributedLockCoreTestCases.cs:446]: Why was this change needed?

Comment (skomis-mm) [reply DistributedLock.Tests/AbstractTestCases/DistributedLockCoreTestCases.cs:446]: @madelson  The need to run `RunLockTaker` / `DistributedLockTaker` with global lock name (prefix + local name) in order to pass cross process tests (`DistributedLockTaker` know nothing about lock prefix, need to pass it somehow).

File: DistributedLock.Tests/Infrastructure/Redis/TestingRedisDatabaseProvider.cs (hunk 4 above)
Comment (skomis-mm) [reply DistributedLock.Tests/Infrastructure/Redis/TestingRedisDatabaseProvider.cs:38]: The need to run RunLockTaker / DistributedLockTaker with global lock name (prefix + local name) in order to pass cross process tests.

File: DistributedLock.Tests/Infrastructure/Redis/TestingRedisSynchronizationStrategy.cs (hunk 3 above)
Comment (skomis-mm) [reply DistributedLock.Tests/Infrastructure/Redis/TestingRedisSynchronizationStrategy.cs:25]: The need to run RunLockTaker / DistributedLockTaker with global lock name (prefix + local name) in order to pass cross process tests.

File: DistributedLock.Tests/Infrastructure/TestingLockProvider.cs (hunk 2 above)
Comment (skomis-mm) [reply DistributedLock.Tests/Infrastructure/TestingLockProvider.cs:22]: The need to run RunLockTaker / DistributedLockTaker with global lock name (prefix + local name) in order to pass cross process tests.

File: DistributedLock.Tests/Infrastructure/TestingSynchronizationStrategy.cs (hunk 1 above)
Comment (skomis-mm) [reply DistributedLock.Tests/Infrastructure/TestingSynchronizationStrategy.cs:24]: The need to run RunLockTaker / DistributedLockTaker with global lock name (prefix + local name) in order to pass cross process tests.
